pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -111,16 +111,16 @@ def __init__(self, argv):
         self.num_of_harts = argv.num_of_harts
         self.fix_sp = argv.fix_sp
         self.use_push_data_section = argv.use_push_data_section
-        self.boot_mode_opts = argv.boot_mode_opts
+        self.boot_mode_opts = argv.boot_mode
 
-        if(self.boot_mode_opts):
+        if self.boot_mode_opts:
             logging.info("Got boot mode option - %0s", self.boot_mode_opts)
-            if(self.boot_mode_opts == "m"):
-                self.init_privileged_mode = privileged_mode_t.MACHINE_MODE.name
-            elif(self.boot_mode_opts == "s"):
-                self.init_privileged_mode = privileged_mode_t.SUPERVISOR_MODE.name
-            elif(self.boot_mode_opts == "u"):
-                self.init_privileged_mode = privileged_mode_t.USER_MODE.name
+            if self.boot_mode_opts == "m":
+                self.init_privileged_mode = privileged_mode_t.MACHINE_MODE
```

Comment:
`self.init_privileged_mode` should be from type `Enum` and not `str`. @aneels3, please confirm.

Reply:
This looks good.
We have to change the string comparison in the whole flow. for eg 
`if "USER_MODE" == self.init_privileged_mode:` this won't work.
I am thinking to create a separate PR for this change such that there won't be any conflicts within the generator.

You can definitely move forward with these changes. I will resolve the possible conflicts in the next PR.

Reply:
Sounds great, thanks @aneels3